### ISSUE TITLE
Feature: replace Adam with AdamW in TSMixer

### ISSUE THREAD

**Issue by davorrunje** [airtai/tpk#32]
## AdamW

wd for AdamW can be found with LR finder or can be a hyperparameter

https://www.fast.ai/posts/2018-07-02-adam-weight-decay.html#adamw

## All the other parameters matter
The way we tune all the other hyper-parameters of the model will impact the best learning rate. That's why when we run the Learning Rate Finder, it's very important to use it with the exact same conditions as during our training. For instance different batch sizes or weight decays will impact the results:

LR Finder for various weight decay values
This can be useful to set some hyper-parameters. For instance, with weight decay, Leslie's advice is to run the learning rate finder for a few values of weight decay, and pick the largest one that will still let us train at a high maximum learning rate. This is how we can come up with the 10−4
 used in our experiments.

In his opinion, the batch size should be set to the highest possible value to fit in the available memory. Then the other hyper-parameters we may have (dropout for instance) can be tuned the same way as weight decay, or just by trying on a cycle and see the results they give. The only thing is to never forget to re-run the Learning Rate Finder, especially when deciding to pick a strategy with an aggressive learning rate close to the maximum possible value.

Training with the 1cycle policy at high learning rates is a method of regularization in itself, so we shouldn't be surprised if we have to reduce the other forms of regularization we were previously using when we put it in place. It will however be more efficient, since we can train for a long time at large learning rates.

https://sgugger.github.io/the-1cycle-policy.html
